src/libs/ajax/fetch/fetch-core.test.ts

```diff
@@ -0,0 +1,118 @@
+import { delay } from '@terra-ui-packages/core-utils';
+import { FetchFn } from '@terra-ui-packages/data-client-core';
+
+import { makeWithRetryOnError, withErrorRejection, withRetry, withRetryOnError } from './fetch-core';
+
+describe('withRetryOnError', () => {
+  // These tests avoid withFakeTimers or useFakeTimers since that setup cause lots of headaches with
+  // the layered promise mechanics under test.
+
+  // use to have shouldNotRetryFn arg always return false
+  const alwaysRetryError = (_error: unknown) => false;
+
+  it('fails after max retries', async () => {
+    // Arrange
+    const fetchFunction: FetchFn = jest.fn(async () => {
+      await delay(51);
+      throw Error('BOOM!');
+    });
+
+    // use faster version of withRetryOnError so that test completes before default 5000ms timout
+    const fasterWithRetryOnError = makeWithRetryOnError({
+      maxTimeout: 2500,
```

Comment:
Is it possible to reduce these intervals further?

Reply:
these values are just preserving what was in existing implementation, just creating the make function to let tests or other future usage decide to use different values.

Reply:
But how long does the test take to run then? The comment "use faster version…" makes me think that the numbers were reduced so that the test would complete within 5 seconds, but I would think we would want it to be significantly faster.

(The variable should also be renamed)

Reply:
renamed... yeah, it was reduced to run within 5 seconds, I wanted to keep it relatively realistic.... We don't have a lot of tests like this, so I thought it was ok to have it take about 3 seconds to wrap up.